MREUnityRuntime/MREUnityRuntimeLib/Behaviors/Contexts/BehaviorContextFactory.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using MixedRealityExtension.API;
+using MixedRealityExtension.App;
+using MixedRealityExtension.Core.Interfaces;
+using System;
+using System.Collections.Generic;
+
+using FactoryFunc = System.Func<
+	MixedRealityExtension.Core.Interfaces.IActor,
+	System.WeakReference<MixedRealityExtension.App.MixedRealityExtensionApp>,
+	MixedRealityExtension.Behaviors.Contexts.BehaviorContextBase>;
+
+namespace MixedRealityExtension.Behaviors.Contexts
+{
+	internal class BehaviorContextFactory
+	{
+		private static Dictionary<BehaviorType, FactoryFunc> s_factoryMethods =
+			new Dictionary<BehaviorType, FactoryFunc>();
+
+		static BehaviorContextFactory()
+		{
+			s_factoryMethods.Add(BehaviorType.Target, CreateTargetBehaviorContext);
+			s_factoryMethods.Add(BehaviorType.Button, CreateButtonBehaviorContext);
+			s_factoryMethods.Add(BehaviorType.Pen, CreatePenBehaviorContext);
+		}
+
+		internal static BehaviorContextBase CreateContext(BehaviorType behaviorType, IActor actor, WeakReference<MixedRealityExtensionApp> appRef)
+		{
+			if (s_factoryMethods.TryGetValue(behaviorType, out FactoryFunc factoryMethod))
+			{
+				return factoryMethod.Invoke(actor, appRef);
+			}
+
+			MixedRealityExtensionApp app;
```

Comment:
(minor) `app` could be declared inline.

Reply:
This was a copy of code written before we updated to a more modern version of C# in the project.  I will update now.

---
In reply to: [428863860](https://github.com/microsoft/mixed-reality-extension-unity/pull/160#discussion_r428863860) [](ancestors = 428863860)